pyteal/ast/ecdsa.py

```diff
@@ -111,33 +114,39 @@ def EcdsaVerify(
     return EcdsaVerifyExpr(curve, data, sigA, sigB, pubkey[0], pubkey[1])
 
 
-def EcdsaDecompress(curve: EcdsaCurve, compressed_pk: Expr) -> MultiValue:
+class EcdsaDecompress(MultiValue):
     """Decompress an ECDSA public key.
 
     Args:
         curve: Enum representing the ECDSA curve used for the public key
         compressed_pk: The compressed public key. Must be 33 bytes long and big endian encoded.
 
     Returns:
-        A MultiValue expression representing the two components of the public key, big endian
+        A MultiValue instance representing the two components of the public key, big endian
         encoded.
     """
 
-    if not isinstance(curve, EcdsaCurve):
-        raise TealTypeError(curve, EcdsaCurve)
+    def __init__(self, curve: EcdsaCurve, compressed_pk: Expr):
+        if not isinstance(curve, EcdsaCurve):
+            raise TealTypeError(curve, EcdsaCurve)
 
-    require_type(compressed_pk, TealType.bytes)
-    return MultiValue(
-        Op.ecdsa_pk_decompress,
-        EcdsaPubkey,
-        immediate_args=[curve.arg_name],
-        args=[compressed_pk],
-    )
+        self.curve = curve
 
+        require_type(compressed_pk, TealType.bytes)
+        super().__init__(
+            Op.ecdsa_pk_decompress,
+            EcdsaPubkey,
+            immediate_args=[curve.arg_name],
+            args=[compressed_pk],
+        )
 
-def EcdsaRecover(
-    curve: EcdsaCurve, data: Expr, recovery_id: Expr, sigA: Expr, sigB: Expr
-) -> MultiValue:
+    def __teal__(self, options: "CompileOptions"):
+        verifyFieldVersion(self.curve.arg_name, self.curve.min_version, options.version)
+
+        return super().__teal__(options)
+
+
+class EcdsaRecover(MultiValue):
```

Comment:
@jdtzmn and reviewers:  Can we treat the method conversion to class as a non-breaking change?

* It _feels_ transparent though I'm not convinced.  For example, a `type` check is liable to fail due to the change in return type (https://stackoverflow.com/questions/1549801/what-are-the-differences-between-type-and-isinstance).
* A few options exist:
  * Accept as is without further change.
  * Accept as is + prefix PR title with _Breaking change:_ (or similar).  Since we use https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes to generate release notes, the disclaimer will be explicit.  
  * Modify `MultiValue` to support field version checks.  There's likely several approaches here.  One avenue is an optional constructor parameter accepting a callback function (`Callable[CompileOptions, None]]`) to be invoked in `__teal__`.
* I'll follow the group's opinion.  My somewhat loosely held opinion is to either prefix PR title or enrich `MultiValue`.  It feels unlikely to me that we'll pattern match `MutliValue` subclasses, so I'm lightly biased towards enriching `MultiValue`.

Reply:
While it will take more work, I prefer to "modify `MultiValue` to support field version checks". The typical usage pattern of MultiValue isn't to extend it unless new fields/methods are needed, like `MaybeValue`'s `hasValue` and `value` methods. This allows for more lightweight implementations of new expressions that use `MultiValue`.

Reply:
Done!

Reply:
@jdtzmn Thanks for exposing `compile_check` - Resolving.